src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -2647,38 +2661,39 @@ def _process_common_blocks(nodes, psyir_parent):
-                    for cb_object in node.children[0]:
-                        for symbol_name in cb_object[1].items:
-                            sym = psyir_parent.symbol_table.lookup(
-                                        str(symbol_name))
-                            if sym.initial_value:
-                                # This is C506 of the F2008 standard.
-                                raise NotImplementedError(
-                                    f"Symbol '{sym.name}' has an initial value"
-                                    f" ({sym.initial_value.debug_string()}) "
-                                    f"but appears in a common block. This is "
-                                    f"not valid Fortran.")
-                            sym.interface = CommonBlockInterface()
-                except KeyError as error:
-                    raise NotImplementedError(
-                        f"The symbol interface of a common block variable "
-                        f"could not be updated because of {error}.") from error
+            if not isinstance(node, Fortran2003.Common_Stmt):
+                continue
+            # Place the declaration statement into a UnknownFortranType
+            # (for now we just want to reproduce it). The name of the
+            # commonblock is not in the same namespace as the variable
+            # symbols names (and there may be multiple of them in a
+            # single statement). So we use an internal symbol name.
+            psyir_parent.symbol_table.new_symbol(
+                root_name="_PSYCLONE_INTERNAL_COMMONBLOCK",
+                symbol_type=DataSymbol,
+                datatype=UnknownFortranType(str(node)))
+
+            # Get the names of the symbols accessed with the commonblock,
+            # they are already defined in the symbol table but they must
+            # now have a common-block interface.
+            try:
+                # Loop over every COMMON block defined in this Common_Stmt
+                for cb_object in node.children[0]:
+                    for symbol_name in cb_object[1].items:
+                        sym = psyir_parent.symbol_table.lookup(
+                                    str(symbol_name))
+                        if sym.initial_value:
+                            # This is C506 of the F2008 standard.
+                            raise NotImplementedError(
+                                f"Symbol '{sym.name}' has an initial value"
+                                f" ({sym.initial_value.debug_string()}) "
+                                f"but appears in a common block. This is "
+                                f"not valid Fortran.")
+                        sym.interface = CommonBlockInterface()
+            except KeyError as error:
+                raise NotImplementedError(
+                    f"The symbol interface of a common block variable "
+                    f"could not be updated because of {error}.") from error
```

Comment:
Isn't this the same? I think I prefer to leave it as it was.

Reply:
This requires a pylint disable, since the nesting is too deep. I often try to reduce nesting level (since I find it easier to read), but I'll fix it up.

Reply:
Ah I see, but I it seems simpler to have positive conditions and no continues.

I also prefer not disabling this kind of pylint warnings, it is a reminder than the code can be improved and is the developer/reviewer who decide what's best. Although we never be able to go to 0 warnings, but I thinks that fine.

Reply:
OK, I reverted that back to the original design, but without the pylint

Reply:
Again, no idea where this ended up. I've changed that back.